File: ruby/bob/bob.rb
class Bob
  def hey(text)
    if text.strip.length == 0
      'Fine. Be that way!'
    elsif text =~ /[A-Z]/ && text.upcase == text
      'Whoa, chill out!'
    elsif text.end_with? '?'
      'Sure.'
    else
      'Whatever.'
    end
  end
end

File: ecmascript/bob/bob.js
class Bob {
  hey(message) {
    if (!message || message.trim().length == 0) {
      return 'Fine. Be that way!';
    } else if (message.endsWith('?')) {
      return 'Sure.';
    } else if (message.endsWith('!')) {
      return 'Whoa, chill out!';
    }
    return 'Whatever.';
  }
}

export default Bob;
